yarn-project/noir-contracts/src/contracts/token_contract/src/main.nr

```diff
@@ -0,0 +1,407 @@
+            AccountContract::at(from.address).is_valid(Context::public(&mut context), message_field);
+        } else {
+            assert(nonce == 0, "invalid nonce");
+        }
+
+        let amount = SafeU120::new(amount);
+        let from_balance = SafeU120::new(storage.public_balances.at(from.address).read()).sub(amount);
+
+        let pending_shields = storage.pending_shields;
+        let mut note = TransparentNote::new(amount.value as Field, secret_hash);
+
+        storage.public_balances.at(from.address).write(from_balance.value as Field);
+        pending_shields.insert_from_public(&mut note);
+        1
+    }
+
+    #[aztec(public)]
+    fn transfer_public(
+        from: AztecAddress,
+        to: AztecAddress,
+        amount: Field,
+        nonce: Field,
+    ) -> Field {
+        let storage = Storage::init(Context::public(&mut context));
+
+        if (from.address != context.msg_sender()) {
+            let selector = compute_selector("transfer_public((Field),(Field),Field,Field)");
+            let message_field = compute_message_hash([context.msg_sender(), context.this_address(), selector, from.address, to.address, amount, nonce]);
+            AccountContract::at(from.address).is_valid(Context::public(&mut context), message_field);
+        } else {
+            assert(nonce == 0, "invalid nonce");
+        }
+
+        let amount = SafeU120::new(amount);
+        let from_balance = SafeU120::new(storage.public_balances.at(from.address).read()).sub(amount);
+        storage.public_balances.at(from.address).write(from_balance.value as Field);
+
+        let to_balance = SafeU120::new(storage.public_balances.at(to.address).read()).add(amount);
+        storage.public_balances.at(to.address).write(to_balance.value as Field);
+    }
+
+    #[aztec(public)]
+    fn burn_public(
+        from: AztecAddress,
+        amount: Field,
+        nonce: Field,
+    ) -> Field {
+        let storage = Storage::init(Context::public(&mut context));
+
+        if (from.address != context.msg_sender()) {
+            let selector = compute_selector("burn_public((Field),Field,Field)");
+            let message_field = compute_message_hash([context.msg_sender(), context.this_address(), selector, from.address, amount, nonce]);
+            AccountContract::at(from.address).is_valid(Context::public(&mut context), message_field);
+        } else {
+            assert(nonce == 0, "invalid nonce");
+        }
+
+        let amount = SafeU120::new(amount);
+        let from_balance = SafeU120::new(storage.public_balances.at(from.address).read()).sub(amount);
+        storage.public_balances.at(from.address).write(from_balance.value as Field);
+
+        let new_supply = SafeU120::new(storage.total_supply.read()).sub(amount);
+        storage.total_supply.write(new_supply.value as Field);
+    }
+
+    #[aztec(private)]
+    fn redeem_shield(
+        to: AztecAddress,
+        amount: Field,
+        secret: Field,
+    ) -> Field {
```

Comment:
hmmm, this says it returns a `Field` but doesnt actually return anything, looks like a bug in aztec-noir.

This appears to happen more on LOTs of methods :(

Reply:
Might be fine to add tests that try returning more than expected as well. Seems like something I would expect to also have issues then